File: test-fixtures/cross-language/python/src/dup_b.py
"""Code-duplication: identical function body to dup_a.py"""


def duplicate_function_b(input_str: str) -> str:
    result = ""
    for ch in input_str:
        if ch.isalpha():
            result += ch.upper()
        elif ch.isdigit():
            result += ch
        else:
            result += "_"
    return result


def another_unique_b(y: int) -> int:
    return y * y + 3 * y + 2

File: test-fixtures/cross-language/typescript/src/dup_b.ts
// Code-duplication: identical function body to dup_a.ts
export function duplicateFunctionB(input: string): string {
  let result = '';
  for (const ch of input) {
    if (/[a-zA-Z]/.test(ch)) {
      result += ch.toUpperCase();
    } else if (/[0-9]/.test(ch)) {
      result += ch;
    } else {
      result += '_';
    }
  }
  return result;
}

export function anotherUniqueB(y: number): number {
  return y * y + 3 * y + 2;
}
